services/blockstorage/sync/state_waiting_for_chunks_test.go

```diff
@@ -0,0 +1,116 @@
+package sync
+
+import (
+	"errors"
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/crypto/keys"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestWaitingMovedToIdleOnTransportError(t *testing.T) {
+	h := newBlockSyncHarness()
+
+	h.storage.When("LastCommittedBlockHeight").Return(primitives.BlockHeight(0)).Times(1)
+	h.gossip.When("SendBlockSyncRequest", mock.Any).Return(nil, errors.New("gossip failure")).Times(1)
+
+	waitingState := h.sf.CreateWaitingForChunksState(h.config.NodePublicKey())
+	nextState := waitingState.processState(h.ctx)
+
+	_, isIdle := nextState.(*idleState)
+
+	require.True(t, isIdle, "expecting back to idle on transport error")
+
+	h.verifyMocks(t)
+}
+
+func TestWaitingMovesToIdleOnTimeout(t *testing.T) {
+	h := newBlockSyncHarness()
+
+	h.storage.When("LastCommittedBlockHeight").Return(primitives.BlockHeight(0)).Times(1)
+	h.gossip.When("SendBlockSyncRequest", mock.Any).Return(nil, nil).Times(1)
+
+	waitingState := h.sf.CreateWaitingForChunksState(h.config.NodePublicKey())
+	nextState := waitingState.processState(h.ctx)
+
+	_, isIdle := nextState.(*idleState)
+
+	require.True(t, isIdle, "expecting back to idle on transport error")
+
+	h.verifyMocks(t)
+}
+
+func TestWaitingAcceptsNewBlockAndMovesToProcessing(t *testing.T) {
+	blocksMessage := builders.BlockSyncResponseInput().Build().Message
+	h := newBlockSyncHarness().withNodeKey(blocksMessage.Sender.SenderPublicKey()) //.withWaitForChunksTimeout(10 * time.Millisecond)
+
+	h.storage.When("LastCommittedBlockHeight").Return(primitives.BlockHeight(10)).Times(1)
+	h.gossip.When("SendBlockSyncRequest", mock.Any).Return(nil, nil).Times(1)
+
+	waitingState := h.sf.CreateWaitingForChunksState(h.config.NodePublicKey())
+	var nextState syncState
```

Comment:
same comment from before about a shared mechanism; more than DRYing up the code, it will force you to give a name to this synchronization mechanism you built here